pkg/social/generic_oauth.go

```diff
@@ -198,6 +225,12 @@ func (s *SocialGenericOAuth) UserInfo(client *http.Client, token *oauth2.Token)
 	name := s.extractName(&data)
 
 	email := s.extractEmail(&data)
+	if email == "" {
+		email, err = s.SearchJSONForEmail(rawUserInfoResponse.Body)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
This should be in `extractEmail` and be checked before email:primary per the doc.